readme.md

```diff
@@ -100,6 +130,38 @@ A list of awesome resources for understanding and applying normalizing flows (NF
 
    > A Jupyter notebook with PyTorch implementations of the most commonly used flows: NICE, RNVP, MAF, Glow, NSF.
 
+1. [`pytorch_flows`](https://github.com/acids-ircam/pytorch_flows) by [acids-ircam](https://github.com/acids-ircam)
+   
+   > A great repo with some basic implementations of normalizing flows from scratch.
+
+2. [`nflows`](https://github.com/bayesiains/nflows) by [bayesiains](https://homepages.inf.ed.ac.uk/imurray2/group/)
+    > A suite of most of the SOTA methods using PyTorch. From an ML group in Edinburgh. They created the current SOTA spline flows. Almost as complete as you'll find from a single repo.
+
+3. [`Normalizing Flows - Introduction (Part 1)`](https://pyro.ai/examples/normalizing_flows_i.html) by [pyro.ai](http://pyro.ai/) 
+  
+    > A tutorial about how to use the `pyro-ppl` library (based on PyTorch) to use Normalizing flows. They have some of the SOTA methods including NSF and MAF.
+
+4. [`destructive-deep-learning`](https://github.com/davidinouye/destructive-deep-learning) by [David Inouye](https://www.davidinouye.com/)
```

Comment:
@jejjohnson Just added the backends to each of the codebases listed under `## Others`. After briefly looking at this one, I don't understand how it relates to NFs. Could you explain what you mean by

> An entire suite of iterative methods to normalizing flows.

Reply:
@janosh Yeah sorry, so this is the author's code base for the Deep Density Destructors by Inouye & Ravikumar (2018) paper. It's essentially the same as normalising flows but you converge iteratively instead of fully parameterised. So you're still using bijective transforms, still computing the Jacobian, still checking for the likelihood, and you can still sample and get probability density estimates. The only difference is that you:

1.  compute one layer or block layer (e.g. Marginal Gaussianization + PCA rotation)
2. check for convergence (e.g log-likelihood using the change-of-variables formula)
3. repeat steps 1-2 until convergence.

To be fair, after the new updates, I would put this under author's code instead of a dedicated section as it's a bit misleading.

---
**Table 1** in the [paper](http://proceedings.mlr.press/v80/inouye18a/inouye18a.pdf) has a nice interesting comparison with the traditional NFs.

Reply:
Thanks for the clarification. I'll just add this to the readme.